cache/s3/s3.go

```diff
@@ -141,6 +144,19 @@ func New(config dict.Dicter) (cache.Interface, error) {
 		return nil, err
 	}
 
+	// If a Proxy/Sidecar/etc.. is used, the Host header needs to
+	// be fixed to allow Request Signing to work.
+	// More info: https://github.com/aws/aws-sdk-go/issues/1473
+	req_signing_host := DefaultReqSigningHost
+	req_signing_host, err = config.String(ConfigKeyReqSigningHost, &req_signing_host)
```

Comment:
`req_signing_host := DefaultReqSigningHost` this does nothing. As it's going to get immediately overwritten in the next line:
`req_signing_host, err = config.String(ConfigKeyReqSigningHost, &req_signing_host)` Just do:
`req_signing_host, err := config.String(ConfigKeyReqSigningHost, &req_signing_host)`

Reply:
Hi! I followed what it is done for all the other parameters, I thought it was a style choice. If I try to build with the line you proposed I get:

```
# github.com/go-spatial/tegola/cache/s3
../../cache/s3/s3.go:149:67: undefined: req_signing_host
```

Reply:
Did you change the `=` to an `:=` like I did in my comment? That's why you are getting the error.

Reply:
Nevermind, I see what I missed. disregard my whole comment. I for some reason did not see that `req_signing_host` was being passed into config.String as the default value. 

Disregard the entire comment, and revert back to what you had. That is correct.